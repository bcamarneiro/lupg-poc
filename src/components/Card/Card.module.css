.card {
  @apply border border-solid rounded-lg overflow-hidden relative;
}

.horizontal {
  @apply flex flex-row gap-4 p-2 items-center;

  .content {
    @apply p-0 flex flex-row gap-3 grow items-center justify-between;
  }

  .image {
    @apply size-6 object-cover rounded-md;
  }

  .name {
    @apply w-36;
  }

  .description {
    @apply truncate ml-0 mr-0 w-72;
  }

  .tags {
    @apply ml-10 mr-auto;
  }

  button {
    @apply py-2;
  }
}

.content {
  @apply p-4 flex flex-col gap-[0.625rem];
}

.price {
}

.name {
  @apply text-[18px] font-medium leading-none;
}

.description {
  @apply text-xs -mt-[5px];
}

.tags {
  @apply flex flex-row gap-1;
}

.image {
  @apply w-full h-35 object-cover;
}
