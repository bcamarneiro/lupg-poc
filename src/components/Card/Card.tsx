import { Product } from '@types';

import styles from './Card.module.css';
import Button from '@components/Button';
import clsx from 'clsx';
import Tag from '@components/Tag';
import formatPrice from '@utils/formatPrice';

interface CardProps {
  product: Product;
  variant?: 'default' | 'horizontal';
}

function Card({ product, variant = 'default' }: CardProps) {
  return (
    <div
      className={clsx(
        styles.card,
        variant === 'horizontal' && styles.horizontal,
        'border-levelup-200 dark:border-levelup-400',
      )}
    >
      <img
        className={styles.image}
        src={product.images[0].url}
        alt={product.images[0].alt}
      />
      <div className={clsx(styles.content, 'bg-levelup-600 dark:bg-levelup-200')}>
        {variant === 'default' && (
          <div className={styles.price}>{formatPrice(product.price.amount, product.price.currency)}</div>
        )}
        <div className={styles.name}>{product.name}</div>
        <div className={clsx(styles.description, 'text-levelup-300 dark:text-levelup-500')}>{product.description}</div>
        <div className={styles.tags}>
          {product.tags.map((tag) => (
            <Tag key={tag}>{tag}</Tag>
          ))}
        </div>
        {variant === 'horizontal' && (
          <div className={styles.price}>{formatPrice(product.price.amount, product.price.currency)}</div>
        )}
        <Button
          className="mt-1"
          onClick={function (): void {
            throw new Error('Function not implemented.');
          }}
        >
          View Details
        </Button>
      </div>
    </div>
  );
}

export default Card;
