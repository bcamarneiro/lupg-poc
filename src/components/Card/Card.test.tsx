import { render, screen } from '@testing-library/react';
import Card from './Card';
import { Product } from '@types';

const mockProduct: Product = {
  uuid: '1',
  name: 'Test Product',
  description: 'This is a test product',
  price: { amount: 10, currency: 'USD' },
  images: [{ url: 'test-image.jpg', alt: 'Test Image' }],
  tags: ['tag1', 'tag2'],
};

describe('Card component', () => {
  it('renders product information correctly', () => {
    const { asFragment } = render(<Card product={mockProduct} />);
    expect(asFragment()).toMatchSnapshot();
    expect(screen.getByText('Test Product')).toBeInTheDocument();
    expect(screen.getByText('This is a test product')).toBeInTheDocument();
    expect(screen.getByText('$10.00')).toBeInTheDocument();
    expect(screen.getByAltText('Test Image')).toBeInTheDocument();
    expect(screen.getByText('tag1')).toBeInTheDocument();
    expect(screen.getByText('tag2')).toBeInTheDocument();
  });

  it('applies horizontal variant correctly', () => {
    render(
      <Card
        product={mockProduct}
        variant="horizontal"
      />,
    );
    const card = screen.getByRole('img').closest('div');
    expect(card?.className).includes('horizontal');
  });

  it('renders "View Details" button', () => {
    render(<Card product={mockProduct} />);
    expect(screen.getByText('View Details')).toBeInTheDocument();
  });
});
