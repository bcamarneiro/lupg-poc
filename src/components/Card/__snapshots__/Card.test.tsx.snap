// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Card component > renders product information correctly 1`] = `
<DocumentFragment>
  <div
    class="_card_1f2933 border-levelup-200 dark:border-levelup-350 bg-levelup-600 dark:bg-levelup-200"
  >
    <img
      alt="Test Image"
      class="_image_1f2933"
      src="test-image.jpg"
    />
    <div
      class="_content_1f2933"
    >
      <div
        class="_price_1f2933"
      >
        $10.00
      </div>
      <div
        class="_name_1f2933"
      >
        Test Product
      </div>
      <div
        class="_description_1f2933 text-levelup-300 dark:text-levelup-500"
      >
        This is a test product
      </div>
      <div
        class="_tags_1f2933"
      >
        <span
          class="_tag_f783ef bg-levelup-500 dark:bg-levelup-300"
        >
          tag1
        </span>
        <span
          class="_tag_f783ef bg-levelup-500 dark:bg-levelup-300"
        >
          tag2
        </span>
      </div>
      <button
        class="mt-1 _button_2a1efe border-levelup-100 dark:border-levelup-800"
      >
        View Details
      </button>
    </div>
  </div>
</DocumentFragment>
`;
